mlir/include/mlir/Dialect/Vector/Transforms/VectorTransforms.h

```diff
@@ -59,6 +59,8 @@ struct VectorTransformsOptions {
     vectorTransferSplit = opt;
     return *this;
   }
+
+  bool useShapeCast = true;
```

Comment:
Could you add a builder method and doc comment, like the other options:
```suggestion
  /// Option to control if vector.transpose can lower to a vector.shape_cast.
  bool useShapeCast = true;
  VectorTransformsOptions &setUseShapeCast(bool opt = true) {
    useShapeCast = opt;
    return *this;
  }
```